bin/map_tools/outliers_process.py

```diff
@@ -49,113 +49,115 @@ def outliers_parser(
     mge_counter = 0
     co_repeats = {}
 
-    # Saving integrated mges in the mge_data
-    integrated_dic = mge_data_parser(mge_data)
-
-    # Parsing the compositional outliers (CO) file
-    # There is one line per flanking repeat. That means two lines per CO
-    # contig_1	62056	62067	contig_1_001:62056-67629|direct_1|GGCGCGCGGCC|gc:3.67|kmer:2.63|combined:3.67|conf:1.000
-    # contig_1	67618	67629	contig_1_001:62056-67629|direct_2|GGCGCGCGGCC|gc:3.67|kmer:2.63|combined:3.67|conf:1.000
-
-    with open(comp_bed, "r") as input_table:
-        lines = (
-            line.rstrip().split("\t")
-            for line in input_table
-            if not line.startswith("#")
-        )
-        for line1, line2 in zip(lines, lines):
-            # Collecting repeats info
-            contig, repeat1_start, repeat1_end, co_info1 = line1
-            coord1 = (int(repeat1_start), int(repeat1_end))
-            co_id, repeat_type1, repeat_seq1, gc, kmer, score, conf = co_info1.split(
-                "|"
-            )
-            metainfo1 = [
-                repeat_type1.replace("direct_1", "DR").replace("inverted_1", "IR"),
-                repeat_seq1,
-                coord1,
-            ]
-
-            contig, repeat2_start, repeat2_end, co_info2 = line2
-            coord2 = (int(repeat2_start), int(repeat2_end))
-            co_id, repeat_type2, repeat_seq2, gc, kmer, score, conf = co_info2.split(
-                "|"
+    if len(comp_bed) > 0:
+        # Saving integrated mges in the mge_data
+        integrated_dic = mge_data_parser(mge_data)
+
+        # Parsing the compositional outliers (CO) file
+        # There is one line per flanking repeat. That means two lines per CO
+        # contig_1	62056	62067	contig_1_001:62056-67629|direct_1|GGCGCGCGGCC|gc:3.67|kmer:2.63|combined:3.67|conf:1.000
+        # contig_1	67618	67629	contig_1_001:62056-67629|direct_2|GGCGCGCGGCC|gc:3.67|kmer:2.63|combined:3.67|conf:1.000
+
```

Comment:
General comment.. I think we need to start coding more defensibely on these kind of parsers. The code assumes a lot about the structure of the GFF, any mistakes upstream and go undetected as there are no checks (empty spaces, missing bits.. ).
I think using some of the AI agentic tools can be very useful to review and improve this code.

This is out of the scope of this PR BTW but before the end of Finding Pheno - related JIRA task https://embl.atlassian.net/browse/EMG-8389